data/basics/userBalance.go

```diff
@@ -379,6 +380,16 @@ type AssetParams struct {
 	Clawback Address `codec:"c"`
 }
 
+func (app AppIndex) ToBeHashed() (protocol.HashID, []byte) {
+	buf := make([]byte, 8)
```

Comment:
just a wild idea - the address we're going to pick is a pseudo-account public address, right ?
if so, can we pick an address that has no private key ?

Reply:
I think we're just putting our faith in the magic of cryptographic hashes, right?  The intent is that the newly allocated address does not have a private key (or, equivalently, such a key exists, but nobody can find it).  It's the same reasoning behind logicsig's controlling an address that we certainly hope does already belong to somebody.

I wrote all that before really realizing what you meant - I'll leave it in case it's helpful to someone else.  I now realize you're asking if, upon app creation, maybe we can pick our app address, as long as the account is not in use.  I think that's insecure because I may know your public address before you fund it (this would be particularly true of a templatized logic sig that has not been instantiated yet, it would also be true if I watch the transaction pool).  I should not be able to make my app use that address, or else when you "come into existence" the app will control your funds.

As for setting the address.  I think the rule you would want is that you can only give the app authority over an account that you yourself have authority.  That is, you certainly don't want to reset an app address to a big foundation address and ask it for money! So I believe that we accomplish this by approaching it from the opposite direction.  We allow you to rekey *your* account to the app's address. And then it has authority. But, of course, you can only rekey that account because you own it.

Reply:
I think @tsachiherman is actually asking about choosing an invalid ed25519 curve point such that no private key exists that could match the address. I believe the MainNet fee sink and rewards pool are invalid points, though I'm not sure how to derive such an invalid point.

Reply:
Correct - what I meant was to select a ed25519 which no private key exists for. This is a post-quantum guarantee equivalent, since no crypto operation would generate that point. The x coordinates on ed25519 curve are mostly on the positive side; in fact, if you'll pick a number on the negative side that is smaller than -486662, I believe that there won't be any solution for that. ( i.e. that number would not be on the curve.. )

Reply:
I had no idea that the public key was used so directly - I sort of assumed it seeded a PRNG that was then used to pick points. I assume by "negative", you mean the 32 bytes is treated as a two-complement u256. So really we could pretty just flip the high bit?

Reply:
I completely agree that we should consult a cryptographer or two here. The randomness you're referring to is used when generating the private key; the public key is a pretty trivial calculation from there : 
```c
int
crypto_sign_ed25519_seed_keypair(unsigned char *pk, unsigned char *sk,
                                 const unsigned char *seed)
{
    ge25519_p3 A;

    crypto_hash_sha512(sk, seed, 32);
    sk[0] &= 248;
    sk[31] &= 127;
    sk[31] |= 64;

    ge25519_scalarmult_base(&A, sk);
    ge25519_p3_tobytes(pk, &A);

    memmove(sk, seed, 32);
    memmove(sk + 32, pk, 32);

    return 0;
}
```

Note that there is another randomness taking place when verifying a signature ( irrelevant in here ).

I do like the idea that you've had of using a hash function, as it "distributes" the account addresses as well, btw.